### ISSUE TITLE
Socket not recognising temporary disruptions

### ISSUE THREAD

**Issue by abhijitkane** [balderdashy/sails.io.js#42]
I'm working on an application that uses the sails.io.js library to make socket calls to a sails server. The calls happen on user events. The socket is connected (io.connect is called) when the app starts.

I have a `on('disconnect')` handler that prevents any messages from being sent through the socket once it's fired. However, I sometimes see this error when I try to send a socket message - 
`Failed to execute 'send' on \'WebSocket\': Still in CONNECTING state.`

If the disconnect handler had been fired, the app wouldn't have attempted to send the message. 

eg. In one user flow, I had to send 20 socket messages to the server - each one after the previous one's response was received. 4 of the messages (not the first 4) couldn't be sent due to this error.

As a workaround, I'm also checking for `socket.transport.websocket.readyState===1` before sending any message. 

Has a similar issue been reported before, or is there any other event I can listen for?

Thanks

**Comment by sailsbot** [comment]
Thanks for posting, @abhijitkane.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
